### PR TITLE
Also show devices under /run/media/$USER on Linux

### PR DESCRIPTION
**Pull request by kerrickstaley**
On Linux, show directories under both `/media` and `/run/media/$USER` in the "Removable Devices" drop-down menu.

udisks2, which some systems use, will auto-mount media into this folder instead of `/media`.

### CODE REVIEW

**Comment by kerrickstaley** [comment]
@daschuer This PR is ready for review. Could you (or another Mixxx maintainer) review it?

Thanks!

**Comment by daschuer** [comment]
Thank you. 

Before merge, we need your permission. 
Please sign https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ
and comment here when done.

File: src/library/browse/browsefeature.cpp
Comment (daschuer) [inline src/library/browse/browsefeature.cpp]: devices_link is unused, please remove.

File: src/library/browse/browsefeature.cpp
Comment (daschuer) [inline src/library/browse/browsefeature.cpp]: This can become a: 
DEBUG_ASSERT(!"Trying to process DEVICE_NODE on macOS");

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank your for taking care. I have only left some minor comments.

File: src/library/browse/browsefeature.h
Comment (uklotzde) [inline src/library/browse/browsefeature.h]: This function should be renamed to `getRemovableDevices()`. Considering the fact that the `this` pointer is just passed through it doesn't even need to be a member function. Move it into the .cpp file enclosed in an anonymous namespace and #ifdefs.

```
QList<TreeItem*> getRemovableDevices(LibraryFeature* pFeature)
```

**Review by uklotzde — CHANGES_REQUESTED** [review]
Don't reveal implementation details in class definition.

File: src/library/browse/browsefeature.cpp
Comment (kerrickstaley) [reply src/library/browse/browsefeature.cpp]: Done.

File: src/library/browse/browsefeature.cpp
Comment (kerrickstaley) [reply src/library/browse/browsefeature.cpp]: Done.

File: src/library/browse/browsefeature.h
Comment (kerrickstaley) [reply src/library/browse/browsefeature.h]: Done.

I renamed the function to `getLinuxRemovableDevices`. I think having `Linux` in the name of the function makes it clearer what the function does. Otherwise, a future reader of the code may think that the function works on all operating systems, whereas it's actually hardcoded with directory paths that are specific to Linux. Let me know if you still prefer that I change it.

**Comment by kerrickstaley** [comment]
@daschuer @uklotzde I addressed comments and filled out the contributor license agreement. Please take a look.

**Review by daschuer — APPROVED** [review]

File: src/library/browse/browsefeature.cpp
Comment (uklotzde) [inline src/library/browse/browsefeature.cpp]: I would clearly separate and try to avoid any dependencies between the conditional compilation sections with #else/#elif.

```
#if defined(__WINDOWS__)
#endif
#if defined(__APPLE__)
#endif
#if defined(__LINUX__)
#endif
```

You have just repeated the pattern from the constructor, that's reasonable. But why not improve the code ;)

File: src/library/browse/browsefeature.cpp
Comment (uklotzde) [inline src/library/browse/browsefeature.cpp]: Please rename the function to just `getRemovableDevices()` and enclose the whole code in `#if defined(__LINUX__)`.

File: src/library/browse/browsefeature.cpp
Comment (uklotzde) [inline src/library/browse/browsefeature.cpp]: The code for windows could also be extracted into a `getRemovableDevices()` function. Finally you can simplify the code as follows:

```
#if defined(__LINUX__) || (__WINDOWS__)
folders += getRemovableDevices(this);
#else
DEBUG_ASSERT(!"Removable devices are not supported for this platform");
#endif

What do you think?

File: src/library/browse/browsefeature.cpp
Comment (kerrickstaley) [reply src/library/browse/browsefeature.cpp]: Done.

File: src/library/browse/browsefeature.cpp
Comment (kerrickstaley) [reply src/library/browse/browsefeature.cpp]: Done.

File: src/library/browse/browsefeature.cpp
Comment (kerrickstaley) [reply src/library/browse/browsefeature.cpp]: I think it's hard to avoid the dependency between the `#if`s here. One idea is that I could also define a `getRemovableDevices` for macOS. Would you like me to do that?

**Comment by kerrickstaley** [comment]
@uklotzde responded to your comments.

**Comment by uklotzde** [comment]
Thanks for improving the code! We don't need to stress this too far, even if I still see some options.

LGTM. Waiting for the CI builds to succeed.

**Review by uklotzde — APPROVED** [review]

**Pull request merged**
